pkg/lib/common.ts

```diff
@@ -26,32 +28,32 @@ export function in_array(array, val) {
     return false;
 }
 
-export function is_function(x) {
+export function is_function(x: unknown): x is (...args: unknown[]) => unknown {
     return typeof x === 'function';
 }
 
-export function is_object(x) {
+export function is_object(x: unknown): x is object {
```

Comment:
is this return type some funky specialization of `boolean`? Does tsc not treat `null` as an object then? If this tries to compute the return value statically, it'd better be correct..

Reply:
See https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
Right, but that doesn't explain if `null` is compatible with `object`. Incidentally, how does one quickly test such a thing?